arangod/Cache/PlainCache.cpp

```diff
@@ -196,10 +181,10 @@ std::shared_ptr<Cache> PlainCache::create(Manager* manager, std::uint64_t id,
                                       enableWindowedStats);
 }
 
-PlainCache::PlainCache(Cache::ConstructionGuard guard, Manager* manager,
+PlainCache::PlainCache(Cache::ConstructionGuard /*guard*/, Manager* manager,
```

Comment:
`ConstructionGuard` is a private/protected subclass of `Cache`.
I think the ConstructionGuard is passed as an argument here so that no cache instances are created from the outside, but only via a factory function. I think it is not necessary to do it like this, and I think I even removed the `ConstructionGuard` from the `TransactionalCache` implementation's ctor. Will do the same for `PlainCache` too.

Reply:
Making the ctors protected/private should be enough here.

Reply:
Ok, currently the ctors need to be public because Cache objects are created using std::make_shared.
So in order to have everyone use the factory function to create cache objects, this `ConstructionGuard` has been added.
It is obviously possible to also do this differently.

Reply:
Ok, got it. Just wanted to clarify this, no need to change anything!